### PR TITLE
Add GitHub Actions, remove Travis CI

### PR DESCRIPTION
**Pull request by olleolleolle**
This PR introduces **GitHub Actions** for CI and for Publish (which is run on GitHub Releases being published).

It removes Travis CI configuration and badge.

## ~TODO~ Steps

- [x] Take `.github/workflows/` from https://github.com/lostisland/faraday/tree/master/.github/workflows  
- [x] Cut out the linting step until we have that in place
- [x] Secrets: supply a `CC_TEST_REPORTER_ID` ENV variable
- [x] Decide if `cane` the rake task should be a part of the CI run. Decision: No, it has been outrun by RuboCop et c.
- [x] Add a badge of GitHub Actions status

See #200

### CODE REVIEW

**Comment by olleolleolle** [comment]
https://github.com/lostisland/faraday_middleware/pull/203 has some linting. 

Perhaps we can do without the RuboCop step before we have that running?

File: README.md [hunk 1]
```diff
@@ -1,7 +1,6 @@
 Faraday Middleware
 ==================
 [![Gem Version](https://badge.fury.io/rb/faraday_middleware.svg)](https://rubygems.org/gems/faraday_middleware)
-[![Build Status](https://travis-ci.org/lostisland/faraday_middleware.svg)](https://travis-ci.org/lostisland/faraday_middleware)
```
Comment (iMacTia) [inline README.md]: Would it be possible to have the GitHub Actions CI badge like on Faraday 😄?

Comment (olleolleolle) [reply README.md]: Probably: I'll add it to the list of things!

Comment (olleolleolle) [reply README.md]: There, added a SVG badge.

**Comment by iMacTia** [comment]
This looks good to go! However merging this before the RuboCop quest is completed will cause linting to be removed from CI, so keeping this on hold until then

**Pull request merged**